### PR TITLE
fix:no warning on empty front template

### LINKED ISSUE #16220: [BUG]: Empty the Front template and click save, no warning
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [X] This bug does not occur in the latest version of Anki

### What are the steps to reproduce this bug?

If I set the Front template empty, no warning shows up.
However, on the desktop version of Anki, it will show a warning if you want to empty the front and don't allow the user to empty the front.
I think we should do the same things that don't allow users to empty the front, the same as the desktop version of Anki

Steps to Reproduce this bug:
1. click card brower
2. click to edit one card
3. click Cards to edit the template
4. clear the text in front template
5. click save

### Expected behaviour

It should not allow the user to empty the Front, just as the desktop Anki does.
image

### Actual behaviour

It allow the user to empty front template and save.
If you review this card,  it will display like this:
image

### Debug info

```text
AnkiDroid Version = 2.17.6 (568538896be943818ca793dd2635822265712def)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 10 (SDK 29)

ProductFlavor = amazon

Manufacturer = unknown

Model = Android SDK built for x86

Hardware = ranchu

Webview User Agent = Mozilla/5.0 (Linux; Android 10; Android SDK built for x86 Build/QSR1.210820.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/74.0.3729.185 Mobile Safari/537.36

ACRA UUID = b8c60fbd-e11e-4dec-9008-73f0d38b53dd

Crash Reports Enabled = true
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by shrutigitte**
## Purpose / Description
Anki and Ankidroid both have same warning dialog when trying to save front card template

## Fixes 
* Fixes #16220 

## How Has This Been Tested?
Google Emulator
image

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 1]
```diff
@@ -518,6 +528,36 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
...
+                showEmptyTemplateDialog()
+                return true
+            }
+            return false
+        }
+
+        private fun showEmptyTemplateDialog() {
+            AlertDialog.Builder(templateEditor).show {
+                templateEditor.tempModel?.let { tempModel ->
+                    val templateCount = tempModel.templateCount
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:544]: If I have two templates, and the problem is on the first template, this should be 1, not 2

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 2]
```diff
@@ -315,6 +324,7 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
         override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
             // Storing a reference to the templateEditor allows us to use member variables
             templateEditor = activity as CardTemplateEditor
+
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:327]: Please revert - PRs shouldn't contain unnecessary spacing changes

**Review by david-allison — CHANGES_REQUESTED** [review]
Thanks!!!

Could you also add a `@NeedsTest` annotation for each testable issue raised in this review

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt [hunk 3]
```diff
@@ -518,6 +528,36 @@ open class CardTemplateEditor : AnkiActivity(), DeckSelectionListener {
             return noteType.isCloze || noteType.isImageOcclusion
         }
 
+        /** Checks if the front of the template is empty or not, and show a dialog if empty
+         * @return true if empty else false **/
+        fun isFrontTemplateEmpty(): Boolean {
+            if (currentTemplate?.front.isNullOrEmpty()) {
```
[the remark this answers is not in the document]
Comment (shrutigitte) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:534]: anki uses update_notetype_legacy and to show errors anki uses AnkiErrors but I couldn't find something as such so I need help as I am a new contributor .

[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:533]: Best to check in Discord to see if anyone has capacity to bring this to completion

[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateEditor.kt:533]: Could you also mark this as a draft if it's not ready for review

**Pull request closed without merging**